mne/connectivity/spectral.py

```diff
@@ -930,21 +932,31 @@ def spectral_connectivity(data, method='coh', indices=None, sfreq=2 * np.pi,
     return con, freqs, times, n_epochs, n_tapers
 
 
-def _prepare_connectivity(epoch_block, tmin, tmax, fmin, fmax, sfreq, indices,
+def _prepare_connectivity(epoch_block, times_in, tmin, tmax,
+                          fmin, fmax, sfreq, indices,
                           mode, fskip, n_bands,
                           cwt_freqs, faverage):
     """Check and precompute dimensions of results data."""
     first_epoch = epoch_block[0]
 
     # get the data size and time scale
-    n_signals, n_times_in, times_in = _get_and_verify_data_sizes(first_epoch)
+    (n_signals, n_times_in,
+     times_in) = _get_and_verify_data_sizes(first_epoch, times=times_in)
 
     if times_in is None:
         # we are not using Epochs or SourceEstimate(s) as input
         times_in = np.linspace(0.0, n_times_in / sfreq, n_times_in,
                                endpoint=False)
 
     n_times_in = len(times_in)
+
+    if tmin is not None and tmin < times_in[0]:
+        warn('start time tmin=%0.2f s outside of the time scope of the data '
```

Comment:
it would be good to test that these warnings are actually getting issued when you expect them to; you can `git grep pytest.warns` for examples of how we normally do this.  This could be in the same test definition as the new test, or in a separate short test, whichever seems cleaner.